src/renku_crac/app.py

```diff
@@ -0,0 +1,369 @@
+"""Compute resource access control (CRAC) app."""
+from dataclasses import dataclass
+from typing import Type, cast
+
+from pydantic import EmailStr
+from sanic import HTTPResponse, Request, Sanic, json
+from sanic_ext import validate
+
+from src import models
+from src.models import errors
+from src.renku_crac.config import Config
+from src.renku_crac.error_handler import CustomErrorHandler
+from src.schemas import apispec
+
+
+@dataclass
+class Server:
+    """Server contains all handlers for CRAC and the configuration."""
+
+    config: Config
+
+    def _get_resource_pool(self):
+        async def handler(request: Request, resource_pool_id: int):
+            res = await self.config.db.get_resource_pools(resource_pool_id, name=request.args.get("name"))
+            if len(res) < 1:
+                raise errors.MissingResourceError(
+                    message=f"The resource pool with id {resource_pool_id} cannot be found."
+                )
+            return json(apispec.ResourcePoolWithId.from_orm(res[0]).dict(exclude_none=True))
+
+        return handler
+
+    def _get_resource_pools(self):
+        async def handler(_: Request):
+            res = await self.config.db.get_resource_pools()
+            return json([apispec.ResourcePoolWithId.from_orm(r).dict(exclude_none=True) for r in res])
+
+        return handler
+
+    def _post_resource_pools(self):
+        @validate(json=apispec.ResourcePool)
+        async def handler(_: Request, body: apispec.ResourcePool):
+            rp = models.ResourcePool.from_dict(body.dict())
```

Comment:
should we check for duplicate names? And add a unique constraint on the DB for name

Reply:
I dont want to enforce unique names for resource pools. We have ids to uniquely determine resource pools. So I want users to be able to name resource pools however they want. For making the k8s resources I will append either the id from the db or a uuid. I think with this in mind it may be better to then use uuid in the endpoints rather than the ids from the database. You already mentioned this in the previous review. I will open a separate PR for this.